### PR TITLE
token-cli: support json output

### PR DESCRIPTION
**Pull request by CriesofCarrots**
As more apps integrate SPL tokens alongside SOL, there is demand for `spl-token-cli` json support that parallels `solana-cli`. This PR adds that functionality.

There is one existing weirdness around `spl-token gc`, since that method may submit multiple transactions. Currently, `spl-token gc --output json` prints multiple json Signature objects (or Signer objects, if `--sign-only`), instead of one nice json array of those objects. Implementing this properly requires an upstream change in the form of `solana_cli_output::return_signers_with_config()` that returns `CliSignOnlyData` instead of strings (or a bunch of duplication). That will follow after the upstream work is implemented. In the meantime, this changeset should meet current needs.

Closes #2128
cc @ryoqun

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
(Sorry, lint should be cleaned in a minute)

**Review by t-nelson — DISMISSED** [review]
Looks great!

wdyt about rejecting `json` for `gc` to avoid introducing something we know will be changed soon?

File: token/cli/src/main.rs [hunk 1]
```diff
@@ -2171,6 +2117,18 @@ fn main() {
...
+            .map(|value| match value {
+                "json" => OutputFormat::Json,
+                "json-compact" => OutputFormat::JsonCompact,
+                _ => unreachable!(),
+            })
+            .unwrap_or(if verbose {
+                OutputFormat::DisplayVerbose
+            } else {
+                OutputFormat::Display
+            });
```
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply token/cli/src/main.rs:2131]: Totes. I can package that with the return_signers api.

**Comment by CriesofCarrots** [comment]
> wdyt about rejecting `json` for `gc` to avoid introducing something we know will be changed soon?

Yep, great idea, will do.

**Pull request merged**